src/pcv/PCVDepositV2.sol

```diff
@@ -0,0 +1,254 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity 0.8.13;
+
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {SafeCast} from "@openzeppelin/contracts/utils/math/SafeCast.sol";
+import {Address} from "@openzeppelin/contracts/utils/Address.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+import {CoreRefV2} from "@voltprotocol/refs/CoreRefV2.sol";
+import {IPCVDepositV2} from "@voltprotocol/pcv/IPCVDepositV2.sol";
+
+/// @title abstract PCV Deposit contract.
+/// if PCV Oracle is not set, this contract will revert and not allow operations
+/// if global reentrancy lock is not set, this contract will still work.
+/// @author Elliot Friedman
+abstract contract PCVDepositV2 is IPCVDepositV2, CoreRefV2 {
+    using SafeERC20 for IERC20;
+    using SafeCast for *;
+
+    /// ------------------------------------------
+    /// --------------- Immutables ---------------
+    /// ------------------------------------------
+
+    /// @notice reference to reward token
+    address public immutable override rewardToken;
+
+    /// @notice reference to underlying token
+    address public immutable override token;
+
```

Comment:
keep profits in the PCV Oracle, keep the last reported balance in the PCV Deposit